core/src/main/java/com/sekwah/advancedportals/core/tags/activation/CommandTag.java

```diff
@@ -0,0 +1,131 @@
+            for (String command : argData) {
+
+                char executionCommand = command.charAt(0);
+                String formattedCommand = command.replaceAll("@player", player.getName());
+
+                switch (executionCommand) {
+                    case '!':
+                        player.performCommand(formattedCommand.substring(1), CommandLevel.OP);
+                        break;
+                    case '#':
+                        player.performCommand(formattedCommand.substring(1), CommandLevel.CONSOLE);
+                        break;
+                    case '^':
+                        player.performCommand(formattedCommand.substring(1), CommandLevel.STAR);
+                        break;
+                    default:
+                        player.performCommand(formattedCommand, CommandLevel.PLAYER);
+                        break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public boolean activated(TagTarget target, PlayerContainer player, ActivationData activationData, String[] argData) {
+        return true;
+    }
+
+    @Override
+    public boolean created(TagTarget target, PlayerContainer player, String[] argData) {
+        if(argData != null) {
+            for (String command : argData) {
+                char executionCommand = command.charAt(0);
+                return switch (executionCommand) {
+                    case '!' -> {
+                        if (!player.hasPermission("advancedportals.createportal.commandlevel.op")) {
+                            player.sendMessage(Lang.translate("tag.command.nopermission")
+                                    .replaceAll("@CommandLevel", "OP"));
+                            yield false;
+                        }
+                        yield true;
+                    }
+                    case '#' -> {
+                        if (!player.hasPermission("advancedportals.createportal.commandlevel.console")) {
+                            player.sendMessage(Lang.translate("tag.command.nopermission")
+                                    .replaceAll("@CommandLevel", "Console"));
+                            yield false;
+                        }
+                        yield true;
+                    }
+                    case '^' -> {
+                        if (!player.hasPermission("advancedportals.createportal.commandlevel.star")) {
+                            player.sendMessage(Lang.translate("tag.command.nopermission")
+                                    .replaceAll("@CommandLevel", "*"));
+                            yield false;
+                        }
+                        yield true;
+                    }
+                    default -> true;
+                };
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void destroyed(TagTarget target, PlayerContainer player, String[] argData) {
+
+    }
+
+    public enum CommandLevel{
+        OP,
+        STAR,
```

Comment:
I think star should be referred to permission not star

Reply:
> I think star should be referred to permission not star
```suggestion
        PERMISSION_WILDCARD,
```
Maybe this instead?

Reply:
yep :D